packages/@react-stately/steplist/src/useStepListState.ts

```diff
@@ -38,25 +38,29 @@ export interface StepListState<T> extends SingleSelectListState<T> {
 export function useStepListState<T extends object>(props: StepListProps<T>): StepListState<T> {
   let state = useSingleSelectListState<T>(props);
 
-  let [lastCompletedStep, setLastCompletedStep] = useControlledState<Key>(props.lastCompletedStep, props.defaultLastCompletedStep, props.onLastCompletedStepChange);
+  let [lastCompletedStep, setLastCompletedStep] = useControlledState<Key | null>(props.lastCompletedStep, props.defaultLastCompletedStep ?? null, props.onLastCompletedStepChange);
```

Comment:
this change is a little funky

i'd have liked to keep undefined through `<Key | undefined>`, however that doesn't match any of the overloads for useControlledState and this is pretty on purpose

as a result of this, I also had to change the signature of onLastcompletedStepChange to also pass null, however, in reality, null will never be passed out

Reply:
I remembered the rest of the reason for this, because I'd thought to call findDefaultKey before this and pass that into the defaultKey spot, however, if we did that, we'd have to call onLastCompletedStepChange manually for initial state, otherwise the user won't get a notification that we've selected a default key.

We could still do this at a later date.